src/Matrix.ts

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2023 MathExpansion
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+//配列(など)をJSONで厳密に比較
+function JSON_equal(x, y) {
+  return JSON.stringify(x) === JSON.stringify(y);
+}
+
+//行列として扱えるか？扱えないならfalse, 扱えるなら[row, col]を返す。
+function is_matrix(array) {
+  const row = array.length;
+  const col = array[0].length;
+  if (col === undefined) {
+    return false;
+  }
+  for (i in array) {
+    if (array[i].length !== col) {
+      return false;
+    }
+  }
+  return [row, col];
+}
+//零行列
+function zeros(row, col) {
+  if (row < 1 || col < 1) {
+    throw new Error('IndexError');
+  }
+  const zeros = [];
+  for (let i = 0; i < row; i++) {
+    const tmp = [];
+    for (let j = 0; j < col; j++) {
+      tmp.push(0);
+    }
+    zeros.push(tmp);
+  }
+  return zeros;
+}
+//単位行列
+function eye(row, col) {
+  if (row < 1 || col < 1) {
+    throw new Error('IndexError');
+  }
+  const eye = [];
+  for (let i = 0; i < row; i++) {
+    const tmp = [];
+    for (let j = 0; j < col; j++) {
+      tmp.push(Number(i === j));
+    }
+    eye.push(tmp);
+  }
+  return eye;
+}
+//対角行列
+function diag(...c) {
+  const diag = [];
+  for (let i = 0; i < c.length; i++) {
+    const tmp = [];
+    for (let j = 0; j < c.length; j++) {
+      if (i !== j) {
+        tmp.push(0);
+      } else {
+        tmp.push(c[i]);
+      }
+    }
+    diag.push(tmp);
+  }
+  return diag;
+}
+//行列の和 A+B
+function add_asmatrix(A, B) {
+  const Amn = is_matrix(A);
+  Bmn = is_matrix(B);
+  if (Amn === false || Bmn === false) {
```

Comment:
なんとなく `is_matrix` の良くなさを感じはしますが。Array は truthy なので `!var` で比較できそうな気がします
```suggestion
  if (!Amn|| !Bmn) {
```